modules/vstudio/_preload.lua

```diff
@@ -195,6 +195,26 @@
 		}
 	}
 
+	p.api.register {
+		name = "enablemodules",
+		scope = { "config" },
+		kind = "string",
+		allowed = {
+			"On",
+			"Off"
+		}
```

Comment:
Using the `boolean` type would probably be better.

Reply:
I thought about using those, but there are also plenty of spots in the code where "Off" and "On" are used in place of a boolean value. Should we start just using boolean instead? If so, that's an easy fix.

Reply:
The inputs for boolean are more flexible:
https://github.com/premake/premake-core/blob/561f9a97d3f68fce1f0340bf33213ba8aa588806/src/base/api.lua#L762-L787

From memory there are issues when using this with look up tables, which is how the toolsets work. So, it's up to you whether you think it's a good idea or not.